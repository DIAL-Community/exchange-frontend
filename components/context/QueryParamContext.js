import { createContext, useState } from 'react'

const QueryParamContext = createContext()
const QueryParamContextProvider = ({ children }) => {
  const [interactionDetected, setInteractionDetected] = useState(false)
  return (
    <QueryParamContext.Provider value={{ ...{ interactionDetected, setInteractionDetected } }}>
      {children}
    </QueryParamContext.Provider>
  )
}

export { QueryParamContextProvider, QueryParamContext }
