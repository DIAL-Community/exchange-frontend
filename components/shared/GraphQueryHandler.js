import { useEffect } from 'react'
import { signIn } from 'next-auth/react'
import { useRouter } from 'next/router'
import { FaCircleExclamation, FaSpinner } from 'react-icons/fa6'
import { FormattedMessage } from 'react-intl'

export const QueryErrorCode = {
  FORBIDDEN: 'FORBIDDEN',
  BAD_REQUEST: 'BAD_REQUEST',
  UNAUTHORIZED: 'UNAUTHORIZED'
}

export const LoadingHandler = () => {
  return (
    <div className='min-h-[75vh] flex items-center justify-center'>
      <div className='flex flex-col gap-y-6'>
        <FaSpinner size='3em' className='spinner mx-auto' />
        <div className='text-center'>
          <FormattedMessage id='ui.general.processing' />
        </div>
      </div>
    </div>
  )
}

export const UnauthorizedErrorHandler = () => {
  useEffect(() => {
    signIn()
  }, [])

  return (
    <div className='min-h-[75vh] flex items-center justify-center'>
      <div className='flex flex-col gap-y-6'>
        <FaSpinner size='3em' className='spinner mx-auto' />
        <div className='text-center'>
          <FormattedMessage id='ui.general.processing' />
        </div>
      </div>
    </div>
  )
}

export const BadRequestErrorHandler = () => {
  const router = useRouter()

  useEffect(() => {
    router.push('/')
  }, [router])

  return (
    <div className='min-h-[75vh] flex items-center justify-center'>
      <div className='flex flex-col gap-y-6'>
        <FaSpinner size='3em' className='spinner mx-auto' />
        <div className='text-center'>
          <FormattedMessage id='ui.general.processing' />
        </div>
      </div>
    </div>
  )
}

export const ForbiddenErrorHandler = () => {
  return (
    <div className='min-h-[75vh] flex items-center justify-center'>
      <div className='flex flex-col gap-y-6'>
        <FaCircleExclamation size='3em' className='spinner mx-auto' />
        <div className='text-center'>
          <FormattedMessage id='ui.general.error.forbidden' />
        </div>
      </div>
    </div>
  )
}

export const GeneralErrorHandler = () => {
  return (
    <div className='min-h-[75vh] flex items-center justify-center'>
      <div className='flex flex-col gap-y-6'>
        <FaCircleExclamation size='3em' className='mx-auto' />
        <div className='text-center'>
          <FormattedMessage id='ui.general.error.general' />
        </div>
      </div>
    </div>
  )
}

export const NotFoundHandler = () => {
  return (
    <div className='min-h-[75vh] flex items-center justify-center'>
      <div className='flex flex-col gap-y-6'>
        <FaSpinner size='3em' className='spinner mx-auto' />
        <div className='text-center'>
          <FormattedMessage id='ui.general.error.notFound' />
        </div>
      </div>
    </div>
  )
}

export const handleQueryError = (error) => {
  if (error?.graphQLErrors) {
    const { graphQLErrors } = error
    for (let e of graphQLErrors) {
      switch (e.extensions.code) {
        case QueryErrorCode.UNAUTHORIZED:
          return <UnauthorizedErrorHandler />
        case QueryErrorCode.FORBIDDEN:
          return <UnauthorizedErrorHandler />
        case QueryErrorCode.BAD_REQUEST:
          return <BadRequestErrorHandler />
        default:
          return <GeneralErrorHandler />
      }
    }
  } else {
    return <GeneralErrorHandler />
  }
}

export const handleLoadingQuery = () => {
  return <LoadingHandler />
}

export const handleMissingData = () => {
  return <NotFoundHandler />
}
