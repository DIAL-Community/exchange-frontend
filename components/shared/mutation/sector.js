import { gql } from '@apollo/client'

export const DELETE_SECTOR = gql`
  mutation DeleteSector($id: ID!) {
    deleteSector(id: $id) {
      sector {
       id
       slug
       name
      }
      errors
    }
  }
`

export const CREATE_SECTOR = gql`
  mutation CreateSector(
    $name: String!
    $slug: String!
    $isDisplayable: Boolean!
    $locale: String!
  ) {
    createSector(
      name: $name
      slug: $slug
      isDisplayable: $isDisplayable
      locale: $locale
    ) {
      sector {
        id
        name
        slug
        locale
        isDisplayable
      }
      errors
    }
  }
`
