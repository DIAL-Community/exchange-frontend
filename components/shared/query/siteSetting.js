import { gql } from '@apollo/client'

export const SITE_SETTING_DETAIL_QUERY = gql`
  query SiteSetting($slug: String) {
    siteSetting(slug: $slug) {
      id
      slug
      name
      description
      faviconUrl
      exchangeLogoUrl
      openGraphLogoUrl
      carouselConfigurations
      heroCardConfigurations
      menuConfigurations
    }
  }
`

export const SITE_SETTINGS_QUERY = gql`
  query SiteSettings {
    siteSettings {
      id
      name
      description
    }
  }
`

export const INITIAL_IMAGE_URL_QUERY = gql`
  query SiteSetting {
    siteSetting {
      id
      faviconUrl
      openGraphLogoUrl
    }
  }
`

export const SITE_SETTING_PAGINATION_ATTRIBUTES_QUERY = gql`
  query PaginationAttributeSiteSetting($search: String) {
    paginationAttributeSiteSetting(search: $search) {
      totalCount
    }
  }
`

export const PAGINATED_SITE_SETTINGS_QUERY = gql`
  query PaginatedSiteSettings(
    $search: String
    $limit: Int!
    $offset: Int!
  ) {
    paginatedSiteSettings(
      search: $search
      offsetAttributes: { limit: $limit, offset: $offset }
    ) {
      id
      name
      slug
      description
    }
  }
`
