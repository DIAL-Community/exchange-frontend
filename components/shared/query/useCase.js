import { gql } from '@apollo/client'

export const USE_CASE_SEARCH_QUERY = gql`
  query UseCases($search: String, $mature: Boolean!) {
    useCases(search: $search, mature: $mature) {
      id
      name
      slug
    }
  }
`

export const USE_CASE_PAGINATION_ATTRIBUTES_QUERY = gql`
  query PaginationAttributeUseCase(
    $search: String
    $sdgs: [String!]
    $showBeta: Boolean
    $showGovStackOnly: Boolean
  ) {
    paginationAttributeUseCase(
      search: $search
      sdgs: $sdgs
      showBeta: $showBeta
      showGovStackOnly: $showGovStackOnly
    ) {
      totalCount
    }
  }
`

export const PAGINATED_USE_CASES_QUERY = gql`
  query PaginatedUseCases(
    $search: String
    $sdgs: [String!]
    $showBeta: Boolean
    $showGovStackOnly: Boolean
    $limit: Int!
    $offset: Int!
  ) {
    paginatedUseCases(
      search: $search
      sdgs: $sdgs
      showBeta: $showBeta
      showGovStackOnly: $showGovStackOnly
      offsetAttributes: { limit: $limit, offset: $offset }
    ) {
      id
      slug
      name
      imageFile
      maturity
      markdownUrl
      sector {
        id
        slug
        name
      }
      govStackEntity
      parsedDescription
      sanitizedDescription
      buildingBlocks {
        id
        slug
        name
      }
      sdgTargets {
        id
        name
      }
    }
  }
`

export const USE_CASE_DETAIL_QUERY = gql`
  query UseCase($slug: String!) {
    useCase(slug: $slug) {
      id
      name
      slug
      sector {
        id
        slug
        name
      }
      maturity
      imageFile
      markdownUrl
      govStackEntity
      useCaseDescription {
        id
        description
      }
      useCaseSteps {
        id
        name
        slug
        workflows {
          id
        }
        products {
          id
        }
        buildingBlocks {
          id
        }
      }
      sdgTargets {
        id
        name
        targetNumber
        sdg {
          id
          name
          slug
          number
          imageFile
        }
      }
      workflows {
        id
        name
        slug
        imageFile
      }
      buildingBlocks {
        id
        name
        slug
        maturity
        category
        imageFile
      }
      tags
    }
  }
`
