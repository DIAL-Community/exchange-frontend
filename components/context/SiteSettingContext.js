import { createContext, useEffect, useState } from 'react'
import { useSession } from 'next-auth/react'
import { useLazyQuery } from '@apollo/client'
import { SITE_SETTING_DETAIL_QUERY } from '../shared/query/siteSetting'

const SiteSettingContext = createContext()
const SiteSettingDispatchContext = createContext()

const SiteSettingProvider = ({ children }) => {
  const { status } = useSession()
  const [updateSiteSettings, { loading, error }] = useLazyQuery(SITE_SETTING_DETAIL_QUERY, {
    onCompleted: (data) => {
      setExchangeLogoUrl(data.siteSetting.exchangeLogoUrl)
      setMenuConfigurations(data.siteSetting.menuConfigurations)
      setCarouselConfigurations(data.siteSetting.carouselConfigurations)
      setHeroCardSection(data.siteSetting.heroCardSection)
      setEnableMarketplace(data.siteSetting.enableMarketplace)
    }
  })

  const [currentStatus, setCurrentStatus] = useState(status === 'authenticated')

  const [exchangeLogoUrl, setExchangeLogoUrl] = useState()
  const [menuConfigurations, setMenuConfigurations] = useState([])
  const [carouselConfigurations, setCarouselConfigurations] = useState([])
  const [heroCardSection, setHeroCardSection] = useState([])

  const [enableMarketplace, setEnableMarketplace] = useState(false)

  const siteSettingValues = {
    exchangeLogoUrl,
    menuConfigurations,
    carouselConfigurations,
    heroCardSection,
    enableMarketplace
  }

  const siteSettingDispatchValues = {
    setExchangeLogoUrl,
    setMenuConfigurations,
    setCarouselConfigurations,
    setHeroCardSection,
    setEnableMarketplace
  }

  useEffect(() => {
    if (status !== currentStatus) {
      setCurrentStatus(status)
      updateSiteSettings()
    }
  }, [status, currentStatus, updateSiteSettings])

  if (loading || error) return null

  return (
    <SiteSettingContext.Provider value={siteSettingValues}>
      <SiteSettingDispatchContext.Provider value={siteSettingDispatchValues}>
        {children}
      </SiteSettingDispatchContext.Provider>
    </SiteSettingContext.Provider>
  )
}

export {
  SiteSettingProvider,
  SiteSettingContext,
  SiteSettingDispatchContext
}
