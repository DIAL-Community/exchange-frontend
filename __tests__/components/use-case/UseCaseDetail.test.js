import { screen } from '@testing-library/dom'
import userEvent from '@testing-library/user-event'
import { QueryParamContextProvider } from '../../../components/context/QueryParamContext'
import { SiteSettingProvider } from '../../../components/context/SiteSettingContext'
import { QueryErrorCode } from '../../../components/shared/GraphQueryHandler'
import { CREATE_USE_CASE } from '../../../components/shared/mutation/useCase'
import { COMMENTS_QUERY } from '../../../components/shared/query/comment'
import { SECTOR_SEARCH_QUERY } from '../../../components/shared/query/sector'
import { DEFAULT_SITE_SETTING_DETAIL_QUERY } from '../../../components/shared/query/siteSetting'
import {
  PAGINATED_USE_CASES_QUERY, USE_CASE_DETAIL_QUERY, USE_CASE_PAGINATION_ATTRIBUTES_QUERY
} from '../../../components/shared/query/useCase'
import UseCaseEdit from '../../../components/use-case/UseCaseEdit'
import { render } from '../../test-utils'
import CustomMockedProvider, { generateMockApolloData } from '../../utils/CustomMockedProvider'
import {
  mockLexicalComponents, mockNextAuthUseSession, mockNextUseRouter, mockTenantApi
} from '../../utils/nextMockImplementation'
import { commentsQuery, createUseCase, sectors, siteSettings, useCaseDetail } from './data/UseCaseDetail.data'
import { paginatedUseCases, useCasePaginationAttribute } from './data/UseCaseMain.data'

mockTenantApi()
mockNextUseRouter()
mockLexicalComponents()
describe('Unit tests for the useCase detail page.', () => {
  const mockUseCase = generateMockApolloData(
    USE_CASE_DETAIL_QUERY,
    {
      'slug': 'remote-learning'
    },
    null,
    useCaseDetail
  )

  const mockSectors = generateMockApolloData(
    SECTOR_SEARCH_QUERY,
    {
      'search': '',
      'locale': 'en'
    },
    null,
    sectors
  )

  const mockUseCaseComments = generateMockApolloData(
    COMMENTS_QUERY,
    {
      'commentObjectId': 190,
      'commentObjectType': 'USE_CASE'
    },
    null,
    commentsQuery
  )

  const mockSiteSetting = generateMockApolloData(
    DEFAULT_SITE_SETTING_DETAIL_QUERY,
    {},
    null,
    siteSettings
  )

  test('Should render detail of a useCase.', async () => {
    const { container } = render(
      <CustomMockedProvider mocks={[mockUseCase, mockSiteSetting, mockUseCaseComments, mockSectors]}>
        <QueryParamContextProvider>
          <SiteSettingProvider>
            <UseCaseEdit slug='remote-learning' />
          </SiteSettingProvider>
        </QueryParamContextProvider>
      </CustomMockedProvider>
    )

    expect(await screen.findByText('Remote Learning')).toBeInTheDocument()
    expect(container).toMatchSnapshot()
  })

  test('Should render unauthorized for non logged in user.', async () => {
    const graphQueryErrors = {
      graphQueryErrors: [{
        'message': 'Viewing is not allowed.',
        'locations': [
          {
            'line': 2,
            'column': 3
          }
        ],
        'path': [
          'buildingBlock'
        ],
        'extensions': {
          'code': QueryErrorCode.UNAUTHORIZED
        }
      }]
    }
    const mockUseCasePolicyQueryError = generateMockApolloData(
      USE_CASE_DETAIL_QUERY,
      {
        'slug': 'remote-learning'
      },
      graphQueryErrors,
      null
    )
    const { container } = render(
      <CustomMockedProvider mocks={[mockUseCasePolicyQueryError, mockSiteSetting, mockUseCaseComments, mockSectors]}>
        <QueryParamContextProvider>
          <SiteSettingProvider>
            <UseCaseEdit slug='remote-learning' />
          </SiteSettingProvider>
        </QueryParamContextProvider>
      </CustomMockedProvider>
    )
    expect(container).toMatchSnapshot()
  })

  test('Should render edit page for logged in user.', async () => {
    mockNextAuthUseSession()

    const mockCreateBuildingBlock = generateMockApolloData(
      CREATE_USE_CASE,
      {
        'name': 'Remote Learning - Edited',
        'slug': 'remote-learning',
        'sectorSlug': 'education-and-social-development-duplicate-0',
        'maturity': 'PUBLISHED',
        'description': 'Description for the use case.',
        'markdownUrl': null,
        'govStackEntity': false
      },
      null,
      createUseCase
    )

    const mockUseCasePaginationAttribute = generateMockApolloData(
      USE_CASE_PAGINATION_ATTRIBUTES_QUERY,
      { search: '', locale: 'en' },
      null,
      useCasePaginationAttribute
    )

    const mockPaginatedUseCases = generateMockApolloData(
      PAGINATED_USE_CASES_QUERY,
      { search: '', locale: 'en', limit: 8, offset: 0 },
      null,
      paginatedUseCases
    )

    const { container } = render(
      <CustomMockedProvider
        mocks={[
          mockSectors,
          mockUseCase,
          mockUseCase,
          mockSiteSetting,
          mockUseCaseComments,
          mockCreateBuildingBlock,
          mockUseCasePaginationAttribute,
          mockPaginatedUseCases
        ]}
      >
        <QueryParamContextProvider>
          <SiteSettingProvider>
            <UseCaseEdit slug='remote-learning' />
          </SiteSettingProvider>
        </QueryParamContextProvider>
      </CustomMockedProvider>
    )

    expect(await screen.findByText('Remote Learning')).toBeInTheDocument()

    const repositoryNameInput = screen.getByDisplayValue('Remote Learning')
    expect(repositoryNameInput.value).toBe('Remote Learning')

    const user = userEvent.setup()
    await user.type(repositoryNameInput, ' - Edited')
    expect(repositoryNameInput.value).toBe('Remote Learning - Edited')

    const repositorySubmitButton = screen.getByText('Submit Use Case')
    await user.click(repositorySubmitButton)

    expect(container).toMatchSnapshot()
  })
})
