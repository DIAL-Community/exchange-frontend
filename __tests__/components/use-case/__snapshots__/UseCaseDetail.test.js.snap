// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit tests for the useCase detail page. Should render detail of a useCase. 1`] = `
<div>
  <div
    class="Toastify"
  />
  <div
    class="lg:px-8 xl:px-56 flex flex-col"
  >
    <div
      class="px-4 lg:px-6 py-4 bg-dial-blue-chalk text-dial-stratos ribbon-detail z-40"
    >
      <div
        class="whitespace-nowrap text-ellipsis overflow-hidden"
      >
        <a
          class="h5"
          href="/"
        >
          Home
        </a>
      </div>
    </div>
    <div
      class="flex flex-col lg:flex-row gap-x-8"
    >
      <div
        class="lg:basis-1/3"
      >
        <div
          class="bg-dial-slate-100 h-full"
        >
          <div
            class="flex flex-col gap-y-3 px-4 lg:px-6 lg:py-3"
          >
            <div
              class="flex flex-col gap-y-4 py-3"
            >
              <div
                class="text-xl text-dial-blueberry font-semibold"
              >
                Remote Learning
              </div>
              <div
                class="flex justify-center items-center py-16 bg-white rounded border"
              >
                <div
                  class="w-20 h-20 px-1 py-1 rounded-full bg-dial-blueberry"
                >
                  <img
                    alt="Logo of Use Case."
                    class="object-contain dial-blueberry-filter w-14 h-14 mx-auto my-2 white-filter"
                    src="http://localhost:3000//assets/use-cases/remote-learning.svg"
                  />
                </div>
              </div>
              <div
                class="flex flex-col gap-y-8 text-sm pt-6 pb-3"
              />
            </div>
            <hr
              class="border-b border-dial-slate-200"
            />
          </div>
        </div>
      </div>
      <div
        class="lg:basis-2/3"
      >
        <div
          class="h-[60vh]"
        >
          <div
            class="flex bg-dial-alice-blue h-full"
          >
            <div
              class="text-dial-stratos text-lg w-full my-auto flex flex-col gap-4"
            >
              <svg
                class="mx-auto"
                fill="currentColor"
                height="3em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 512 512"
                width="3em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M256 512A256 256 0 1 0 256 0a256 256 0 1 0 0 512zm0-384c13.3 0 24 10.7 24 24V264c0 13.3-10.7 24-24 24s-24-10.7-24-24V152c0-13.3 10.7-24 24-24zM224 352a32 32 0 1 1 64 0 32 32 0 1 1 -64 0z"
                />
              </svg>
              <div
                class="text-center mt-5"
              >
                You are not authorized to view this page
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Unit tests for the useCase detail page. Should render edit page for logged in user. 1`] = `
<div>
  <div
    class="Toastify"
  />
  <div
    class="lg:px-8 xl:px-56 flex flex-col"
  >
    <div
      class="px-4 lg:px-6 py-4 bg-dial-blue-chalk text-dial-stratos ribbon-detail z-40"
    >
      <div
        class="whitespace-nowrap text-ellipsis overflow-hidden"
      >
        <a
          class="h5"
          href="/"
        >
          Home
        </a>
      </div>
    </div>
    <div
      class="flex flex-col lg:flex-row gap-x-8"
    >
      <div
        class="lg:basis-1/3"
      >
        <div
          class="bg-dial-slate-100 h-full"
        >
          <div
            class="flex flex-col gap-y-3 px-4 lg:px-6 lg:py-3"
          >
            <div
              class="flex flex-col gap-y-4 py-3"
            >
              <div
                class="text-xl text-dial-blueberry font-semibold"
              >
                Remote Learning
              </div>
              <div
                class="flex justify-center items-center py-16 bg-white rounded border"
              >
                <div
                  class="w-20 h-20 px-1 py-1 rounded-full bg-dial-blueberry"
                >
                  <img
                    alt="Logo of Use Case."
                    class="object-contain dial-blueberry-filter w-14 h-14 mx-auto my-2 white-filter"
                    src="http://localhost:3000//assets/use-cases/remote-learning.svg"
                  />
                </div>
              </div>
              <div
                class="flex flex-col gap-y-8 text-sm pt-6 pb-3"
              />
            </div>
            <hr
              class="border-b border-dial-slate-200"
            />
          </div>
        </div>
      </div>
      <div
        class="lg:basis-2/3"
      >
        <form>
          <div
            class="px-4 lg:px-0 py-4 lg:py-6"
          >
            <div
              class="flex flex-col gap-y-6 text-sm"
            >
              <div
                class="text-xl font-semibold text-dial-blueberry"
              >
                Edit Remote Learning
              </div>
              <div
                class="flex flex-col gap-y-2"
              >
                <label
                  class="required-field text-dial-blueberry"
                  for="name"
                >
                  Name
                </label>
                <input
                  class="text-sm w-full text-sm"
                  id="name"
                  name="name"
                  placeholder="Name"
                  type="text"
                  value=""
                />
              </div>
              <div
                class="flex flex-col gap-y-2"
              >
                <label
                  class="required-field text-dial-blueberry"
                  for="use-case-sector"
                >
                  Sector
                </label>
                <div
                  class=" css-1uaxf62-container"
                  id="use-case-sector"
                >
                  <span
                    class="css-1f43avz-a11yText-A11yText"
                    id="react-select-:r0:-live-region"
                  />
                  <span
                    aria-atomic="false"
                    aria-live="polite"
                    aria-relevant="additions text"
                    class="css-1f43avz-a11yText-A11yText"
                    role="log"
                  />
                  <div
                    class="react-select__control css-afrqkl-control"
                  >
                    <div
                      class="react-select__value-container css-17t7gkf-ValueContainer"
                    >
                      <div
                        class="react-select__placeholder css-5rjclb-placeholder"
                        id="react-select-:r0:-placeholder"
                      >
                        Sector
                      </div>
                      <div
                        class="react-select__input-container css-qbdosj-Input"
                        data-value=""
                      >
                        <input
                          aria-activedescendant=""
                          aria-autocomplete="list"
                          aria-describedby="react-select-:r0:-placeholder"
                          aria-expanded="false"
                          aria-haspopup="true"
                          autocapitalize="none"
                          autocomplete="off"
                          autocorrect="off"
                          class="react-select__input"
                          id="react-select-id-sector"
                          role="combobox"
                          spellcheck="false"
                          style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                          tabindex="0"
                          type="text"
                          value=""
                        />
                      </div>
                    </div>
                    <div
                      class="react-select__indicators css-1hb7zxy-IndicatorsContainer"
                    >
                      <span
                        class="react-select__indicator-separator css-nyxz9h-indicatorSeparator"
                      />
                      <div
                        aria-hidden="true"
                        class="react-select__indicator react-select__dropdown-indicator css-2hgo35-indicatorContainer"
                      >
                        <svg
                          aria-hidden="true"
                          class="text-lg"
                          fill="none"
                          height="1em"
                          stroke="currentColor"
                          stroke-width="2"
                          viewBox="0 0 24 24"
                          width="1em"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M21 21l-6-6m2-5a7 7 0 11-14 0 7 7 0 0114 0z"
                            stroke-linecap="round"
                            stroke-linejoin="round"
                          />
                        </svg>
                      </div>
                    </div>
                  </div>
                </div>
                <p
                  class="text-xs italic text-validation-error"
                >
                  This field is required
                </p>
              </div>
              <div
                class="flex flex-col gap-y-2"
              >
                <label
                  class="required-field text-dial-blueberry"
                  for="use-case-maturity"
                >
                  Maturity
                </label>
                <div
                  class=" css-1uaxf62-container"
                  id="use-case-maturity"
                >
                  <span
                    class="css-1f43avz-a11yText-A11yText"
                    id="react-select-:r1:-live-region"
                  />
                  <span
                    aria-atomic="false"
                    aria-live="polite"
                    aria-relevant="additions text"
                    class="css-1f43avz-a11yText-A11yText"
                    role="log"
                  />
                  <div
                    class="react-select__control css-afrqkl-control"
                  >
                    <div
                      class="react-select__value-container react-select__value-container--has-value css-17t7gkf-ValueContainer"
                    >
                      <div
                        class="react-select__single-value css-1dimb5e-singleValue"
                      >
                        Published
                      </div>
                      <div
                        class="react-select__input-container css-qbdosj-Input"
                        data-value=""
                      >
                        <input
                          aria-activedescendant=""
                          aria-autocomplete="list"
                          aria-expanded="false"
                          aria-haspopup="true"
                          autocapitalize="none"
                          autocomplete="off"
                          autocorrect="off"
                          class="react-select__input"
                          id="react-select-id-maturity"
                          role="combobox"
                          spellcheck="false"
                          style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                          tabindex="0"
                          type="text"
                          value=""
                        />
                      </div>
                    </div>
                    <div
                      class="react-select__indicators css-1hb7zxy-IndicatorsContainer"
                    >
                      <span
                        class="react-select__indicator-separator css-nyxz9h-indicatorSeparator"
                      />
                      <div
                        aria-hidden="true"
                        class="react-select__indicator react-select__dropdown-indicator css-2hgo35-indicatorContainer"
                      >
                        <svg
                          aria-hidden="true"
                          class="text-lg"
                          fill="none"
                          height="1em"
                          stroke="currentColor"
                          stroke-width="2"
                          viewBox="0 0 24 24"
                          width="1em"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M21 21l-6-6m2-5a7 7 0 11-14 0 7 7 0 0114 0z"
                            stroke-linecap="round"
                            stroke-linejoin="round"
                          />
                        </svg>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="flex flex-col gap-y-2"
              >
                <label
                  class="text-dial-sapphire"
                  for="markdown-url"
                >
                  Markdown URL
                </label>
                <div
                  class="border border-transparent url-input bg-white flex items-center"
                >
                  <input
                    class="url w-full text-sm placeholder-dial-gray border-0 py-0 px-0 focus:outline-0"
                    id="markdown-url"
                    placeholder="Markdown URL"
                    value=""
                  />
                </div>
              </div>
              <div
                class="flex flex-col gap-y-2"
              >
                <label
                  class="text-dial-sapphire"
                  for="image-uploader"
                >
                  Logo
                </label>
                <label
                  class="flex items-center"
                >
                  <input
                    accept=".jpg,.jpeg,.png,.bmp"
                    class="bg-white file:hidden w-full text-sm"
                    id="image-uploader"
                    name="imageFile"
                    type="file"
                  />
                  <div
                    class="-ml-11 flex p-1"
                  >
                    <span
                      class="h-full pl-1 border-l text-dial-gray-dark border-dial-gray-dark"
                    >
                      <svg
                        fill="currentColor"
                        height="1em"
                        stroke="currentColor"
                        stroke-width="0"
                        version="1.1"
                        viewBox="0 0 16 16"
                        width="1em"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M13 14h-10v-2l3-5 4.109 5 2.891-2v4z"
                        />
                        <path
                          d="M13 7.5c0 0.828-0.672 1.5-1.5 1.5s-1.5-0.672-1.5-1.5 0.672-1.5 1.5-1.5c0.828 0 1.5 0.672 1.5 1.5z"
                        />
                        <path
                          d="M14.341 3.579c-0.347-0.473-0.831-1.027-1.362-1.558s-1.085-1.015-1.558-1.362c-0.806-0.591-1.197-0.659-1.421-0.659h-7.75c-0.689 0-1.25 0.561-1.25 1.25v13.5c0 0.689 0.561 1.25 1.25 1.25h11.5c0.689 0 1.25-0.561 1.25-1.25v-9.75c0-0.224-0.068-0.615-0.659-1.421zM12.271 2.729c0.48 0.48 0.856 0.912 1.134 1.271h-2.406v-2.405c0.359 0.278 0.792 0.654 1.271 1.134zM14 14.75c0 0.136-0.114 0.25-0.25 0.25h-11.5c-0.135 0-0.25-0.114-0.25-0.25v-13.5c0-0.135 0.115-0.25 0.25-0.25 0 0 7.749-0 7.75 0v3.5c0 0.276 0.224 0.5 0.5 0.5h3.5v9.75z"
                        />
                      </svg>
                    </span>
                  </div>
                </label>
                <div
                  class="text-xs italic text-dial-stratos"
                >
                  Note: Supported file types are jpg, png, and bmp
                </div>
              </div>
              <label
                class="flex gap-x-2 items-center self-start text-dial-sapphire"
              >
                <input
                  class="h-5 w-5"
                  name="govStackEntity"
                  type="checkbox"
                />
                GovStack use case
              </label>
              <div
                class="flex flex-col gap-y-2"
              >
                <label
                  class="text-dial-sapphire required-field"
                >
                  Description
                </label>
                <div
                  class="relative"
                >
                  <svg
                    class="absolute text-lg inset-x-1/2 top-10 spinner"
                    fill="currentColor"
                    height="2em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 512 512"
                    width="2em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M304 48a48 48 0 1 0 -96 0 48 48 0 1 0 96 0zm0 416a48 48 0 1 0 -96 0 48 48 0 1 0 96 0zM48 304a48 48 0 1 0 0-96 48 48 0 1 0 0 96zm464-48a48 48 0 1 0 -96 0 48 48 0 1 0 96 0zM142.9 437A48 48 0 1 0 75 369.1 48 48 0 1 0 142.9 437zm0-294.2A48 48 0 1 0 75 75a48 48 0 1 0 67.9 67.9zM369.1 437A48 48 0 1 0 437 369.1 48 48 0 1 0 369.1 437z"
                    />
                  </svg>
                  <div
                    class="html-editor"
                  >
                    <textarea
                      id="use-case-description-editor"
                      style="visibility: hidden;"
                    />
                  </div>
                </div>
              </div>
              <div
                class="flex flex-wrap text-base mt-6 gap-3"
              >
                <button
                  class="submit-button"
                  type="submit"
                >
                  Submit Use Case
                </button>
                <button
                  class="cancel-button"
                  type="button"
                >
                  Cancel
                </button>
              </div>
            </div>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;

exports[`Unit tests for the useCase detail page. Should render unauthorized for non logged in user. 1`] = `
<div>
  <div
    class="Toastify"
  />
  <div
    class="lg:px-8 xl:px-56 flex flex-col"
  >
    <div
      class="px-4 lg:px-6 py-4 bg-dial-blue-chalk text-dial-stratos ribbon-detail z-40"
    >
      <div
        class="whitespace-nowrap text-ellipsis overflow-hidden"
      >
        <a
          class="h5"
          href="/"
        >
          Home
        </a>
      </div>
    </div>
    <div
      class="flex flex-col lg:flex-row gap-x-8"
    >
      <div
        class="lg:basis-1/3"
      >
        <div
          class="bg-dial-slate-100 h-full"
        >
          <div
            class="flex flex-col gap-y-3 px-4 lg:px-6 lg:py-3"
          >
            <div
              class="flex flex-col gap-y-4 py-3"
            >
              <div
                class="text-xl text-dial-blueberry font-semibold"
              >
                Remote Learning
              </div>
              <div
                class="flex justify-center items-center py-16 bg-white rounded border"
              >
                <div
                  class="w-20 h-20 px-1 py-1 rounded-full bg-dial-blueberry"
                >
                  <img
                    alt="Logo of Use Case."
                    class="object-contain dial-blueberry-filter w-14 h-14 mx-auto my-2 white-filter"
                    src="http://localhost:3000//assets/use-cases/remote-learning.svg"
                  />
                </div>
              </div>
              <div
                class="flex flex-col gap-y-8 text-sm pt-6 pb-3"
              />
            </div>
            <hr
              class="border-b border-dial-slate-200"
            />
          </div>
        </div>
      </div>
      <div
        class="lg:basis-2/3"
      >
        <div
          class="h-[60vh]"
        >
          <div
            class="flex bg-dial-alice-blue h-full"
          >
            <div
              class="text-dial-stratos text-lg w-full my-auto flex flex-col gap-4"
            >
              <svg
                class="mx-auto"
                fill="currentColor"
                height="3em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 512 512"
                width="3em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M256 512A256 256 0 1 0 256 0a256 256 0 1 0 0 512zm0-384c13.3 0 24 10.7 24 24V264c0 13.3-10.7 24-24 24s-24-10.7-24-24V152c0-13.3 10.7-24 24-24zM224 352a32 32 0 1 1 64 0 32 32 0 1 1 -64 0z"
                />
              </svg>
              <div
                class="text-center mt-5"
              >
                You are not authorized to view this page
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
